lib/telemetry_metrics_prometheus/supervisor.ex

```diff
@@ -15,7 +15,16 @@ defmodule TelemetryMetricsPrometheus.Supervisor do
       Plug.Cowboy.child_spec(
         scheme: Keyword.get(args, :protocol),
         plug: {Router, [name: Keyword.get(args, :name)]},
-        options: Keyword.get(args, :options)
+        options:
+          case Keyword.fetch(args, :name) do
```

Comment:
We can omit all the case logic. `:name` is always set in the main file, so we don't need to be guarded. https://github.com/beam-telemetry/telemetry_metrics_prometheus/blob/main/lib/telemetry_metrics_prometheus.ex#L132

```elixir
...,
options: Keyword.put_new(args.options, :ref, args[:name])
```

And since we're specifying a default value, we should probably move it to the main file where we already set defaults. https://github.com/beam-telemetry/telemetry_metrics_prometheus/blob/main/lib/telemetry_metrics_prometheus.ex#L125-L126